### PR TITLE
Add Content Tab to document python and Console SDK

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
Added [Content Tabs](https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#linked-content-tabs-feature-disabled) to document the CLI and python SDK

* The values with defaults were mainly omitted from the documentation 
* The tutorials were left to only include the CLI way of working since this is the recommended approach

### CODE REVIEW

File: docs/guides/first_pipeline.md [hunk 1]
```diff
@@ -10,13 +10,13 @@ Clone the Fondant GitHub repository
...
 
-Make sure that Docker Compose is running, navigate to `fondant/examples/pipelines/filter-cc-25m`, and initiate the pipeline by executing:
+Make sure that Docker Compose is [installed](installation.md), navigate to `fondant/examples/pipelines/filter-cc-25m`, and initiate the pipeline by executing:
 
 ```
 fondant run local pipeline.py
 ```
 
-Note: For local testing purposes, the pipeline will only download the first 10,000 images. If you want to download the full dataset, you will need to modify the component arguments in the pipeline.py file, specifically the following part:
+**Note:** For local testing purposes, the pipeline will only download the first 10,000 images. If you want to download the full dataset, you will need to modify the component arguments in the pipeline.py file, specifically the following part:
```
Comment (RobbeSneyders) [inline docs/guides/first_pipeline.md:19]: FYI, we can add actual notes: https://squidfunk.github.io/mkdocs-material/reference/admonitions

Comment (RobbeSneyders) [reply docs/guides/first_pipeline.md:19]: I see you did this below :)

File: docs/pipeline.md [hunk 2]
```diff
@@ -70,33 +70,131 @@ dependencies between components to build the DAG.
 Currently Fondant supports linear DAGs with single dependencies. Support for non-linear DAGs will be
```
Comment (RobbeSneyders) [inline docs/pipeline.md:70]: This needs to be indented.

File: docs/pipeline.md [hunk 3]
```diff
@@ -70,33 +70,131 @@ dependencies between components to build the DAG.
...
 available in future releases.
 
-## Compiling and Running a pipeline
+## Compiling a pipeline
 
 Once all your components are added to your pipeline you can use different compilers to run your
 pipeline:
 
 !!! note "IMPORTANT"
 When using other runners you will need to make sure that your new environment has access to:
```
Comment (RobbeSneyders) [inline docs/pipeline.md:79]: This note is broken

File: docs/runners/local.md [hunk 4]
```diff
@@ -14,31 +14,98 @@ In order to use the local runner, you need to have a recent version of [docker-c
...
+
+    === "GCP"
+    
+        ```python
+        from fondant.pipeline.compiler import DockerCompiler
+        from fondant.pipeline.runner import DockerRunner
+        from fondant.core.schema import CloudCredentialsMount
+        
+        gcp_mounr_dir = CloudCredentialsMount.GCP.value
+        compiler = DockerCompiler(extra_volumnes=gcp_mounr_dir)
```
Comment (RobbeSneyders) [inline docs/runners/local.md:56]:
```suggestion
        gcp_mount_dir = CloudCredentialsMount.GCP.value
        compiler = DockerCompiler(extra_volumes=gcp_mount_dir)
```

I also think `extra_volumes` needs to be a list.

Same changes in the other tabs.

File: docs/pipeline.md [hunk 5]
```diff
@@ -70,33 +70,131 @@ dependencies between components to build the DAG.
...
-- The base path of your pipeline (as mentioned above)
-- The images used in your pipeline (make sure you have access to the registries where the images are
-  stored)
+    - The base path of your pipeline (as mentioned above)
+    - The images used in your pipeline (make sure you have access to the registries where the images are
+    stored)
+
+=== "Console"
+    
+    === "Local Compiler"
```
Comment (RobbeSneyders) [inline docs/pipeline.md:87]: If we rename this to "Local", "Vertex", and "Kubeflow" for both the compilers and runners, the tabs will be linked.

File: src/fondant/core/schema.py [hunk 6]
```diff
@@ -2,13 +2,26 @@
...
 import re
 import typing as t
+from enum import Enum
 
 import pyarrow as pa
 
 from fondant.core.exceptions import InvalidTypeSchema
 
+
+class CloudCredentialsMount(Enum):
```
Comment (RobbeSneyders) [inline src/fondant/core/schema.py:15]: I think this is only used by the runner SDK, so would move it there.

**Review by RobbeSneyders — COMMENTED** [review]
Thanks @PhilippeMoussalli, looks great!

File: docs/components/publishing_components.md [hunk 7]
```diff
@@ -1 +1,37 @@
-# Coming Soon 
+## Building and publishing components
+
+Before you can use your custom component within your pipeline you have to build the component into a
```
[the remark this answers is not in the document]
Comment (PhilippeMoussalli) [reply docs/components/publishing_components.md:3]: Done :)

File: docs/pipeline.md (hunk 2 above)
Comment (PhilippeMoussalli) [reply docs/pipeline.md:70]: Done :)

File: docs/runners/local.md (hunk 4 above)
Comment (PhilippeMoussalli) [reply docs/runners/local.md:56]: I changed the script to also accept a string (single volume to mount)

File: docs/pipeline.md (hunk 5 above)
Comment (PhilippeMoussalli) [reply docs/pipeline.md:87]: Good point, changed it

File: src/fondant/core/schema.py (hunk 6 above)
Comment (PhilippeMoussalli) [reply src/fondant/core/schema.py:15]: Moved it to the compiler since this is where the extra volume mounting happens

Comment (RobbeSneyders) [reply src/fondant/core/schema.py:15]: Great, I meant pipeline SDK anyway.

File: docs/data_explorer.md [hunk 8]
```diff
@@ -13,19 +13,42 @@ The chosen subset (and the columns within the subset) can be explored in 3 tabs.
...
+    fondant explore --base_path $BASE_PATH
+    ```
+
+=== "Python"
+
+    ```python
+    from fondant.explore import run_explorer_app
+    
+    BASE_PATH = "gs://foo/bar"
+    run_explorer_app(base_path=$BASE_PATH)
```
Comment (mrchtr) [inline docs/data_explorer.md:50]: I guess it should be:
```suggestion
    run_explorer_app(base_path=BASE_PATH)
```

File: docs/data_explorer.md [hunk 9]
```diff
@@ -13,19 +13,42 @@ The chosen subset (and the columns within the subset) can be explored in 3 tabs.
...
+    fondant explore --base_path $BASE_PATH
+    ```
+
+=== "Python"
+
+    ```python
+    from fondant.explore import run_explorer_app
+    
+    BASE_PATH = "your_base_path"
+    run_explorer_app(base_path=$BASE_PATH)
```
Comment (mrchtr) [inline docs/data_explorer.md:28]:
```suggestion
    run_explorer_app(base_path=BASE_PATH)
```

File: docs/runners/local.md [hunk 10]
```diff
@@ -14,31 +14,98 @@ In order to use the local runner, you need to have a recent version of [docker-c
...
-
-NOTE: that the pipeline ref is the path to the compiled pipeline spec OR a reference to a fondant pipeline in which case a Docker compiler will compile the pipeline
-to a docker compose specification before running the pipeline.This will start the pipeline and provide logs per component (service).
+=== "Console"
+
+    ```bash
+    fondant run local <pipeline_ref>
+    ```
+
+    If you want to use remote paths (GCS, S3, etc.) you can use the `--auth-gcp`, `--auth-aws` or `--auth-azure`.
```
Comment (mrchtr) [inline docs/runners/local.md:24]: Maybe it makes sense to split it up as well (similar to the example below). 
Selecting/activate the right session/profile for different cloud providers works differently. 
If we would add the commands for each cloud provider everything would be here on the page.

**Review by mrchtr — COMMENTED** [review]
Thanks @PhilippeMoussalli. Found a few small things.

File: docs/runners/local.md (hunk 4 above)
Comment (PhilippeMoussalli) [reply docs/runners/local.md]: woops fixed it

Comment (RobbeSneyders) [reply docs/runners/local.md]: One more typo remaining :slightly_smiling_face: 

"extra_volumnes" -> "extra_volumes"

**Review by RobbeSneyders — APPROVED** [review]

**Pull request merged**